tensorflow/python/data/kernel_tests/range_test.py

```diff
@@ -142,6 +150,90 @@ def testStopLessThanStartWithNegativeStep(self, output_type):
+                                 parameterized.TestCase):
+
+  def _iterator_checkpoint_prefix_local(self):
+    return os.path.join(self.get_temp_dir(), "iterator")
+
+  def _save_op(self, iterator_resource):
+    iterator_state_variant = gen_dataset_ops.serialize_iterator(
+        iterator_resource)
+    save_op = io_ops.write_file(
+        self._iterator_checkpoint_prefix_local(),
+        parsing_ops.serialize_tensor(iterator_state_variant))
+    return save_op
+
+  def _restore_op(self, iterator_resource):
+    iterator_state_variant = parsing_ops.parse_tensor(
+        io_ops.read_file(self._iterator_checkpoint_prefix_local()),
+        dtypes.variant)
+    restore_op = gen_dataset_ops.deserialize_iterator(iterator_resource,
+                                                      iterator_state_variant)
+    return restore_op
+
+  # TODO(vikoth18): implement eager mode checkpoint tests
+  @combinations.generate(
+      combinations.combine(tf_api_version=1, mode=["graph"]))
+  def testSaveRestore(self):
+
+    def _build_graph(start, stop):
+      iterator = dataset_ops.make_initializable_iterator(
+          dataset_ops.Dataset.range(start, stop))
+      init_op = iterator.initializer
+      get_next = iterator.get_next()
+      save_op = self._save_op(iterator._iterator_resource)
+      restore_op = self._restore_op(iterator._iterator_resource)
+      return init_op, get_next, save_op, restore_op
+
+    # Saving and restoring in different sessions.
+    start = 2
+    stop = 10
+    break_point = 5
+    with ops.Graph().as_default() as g:
+      init_op, get_next, save_op, _ = _build_graph(start, stop)
+      with self.session(graph=g):
+        self.evaluate(variables.global_variables_initializer())
+        self.evaluate(init_op)
+        for i in range(start, break_point):
+          self.assertEqual(i, self.evaluate(get_next))
+        self.evaluate(save_op)
+
+    with ops.Graph().as_default() as g:
+      init_op, get_next, _, restore_op = _build_graph(start, stop)
+      with self.session(graph=g):
+        self.evaluate(init_op)
+        self.evaluate(restore_op)
+        for i in range(break_point, stop):
+          self.assertEqual(i, self.evaluate(get_next))
+        with self.assertRaises(errors.OutOfRangeError):
+          self.evaluate(get_next)
+
+    # Saving and restoring in same session.
+    with ops.Graph().as_default() as g:
+      init_op, get_next, save_op, restore_op = _build_graph(start, stop)
+      with self.session(graph=g):
+        self.evaluate(variables.global_variables_initializer())
+        self.evaluate(init_op)
+        for i in range(start, break_point):
+          self.assertEqual(i, self.evaluate(get_next))
+        self.evaluate(save_op)
+        self.evaluate(restore_op)
+        for i in range(break_point, stop):
+          self.assertEqual(i, self.evaluate(get_next))
+        with self.assertRaises(errors.OutOfRangeError):
+          self.evaluate(get_next)
```

Comment:
Remove this as it is outdated.

Reply:
removed the methods, imports and extra build deps. Test log after change.
```
//tensorflow/python/data/kernel_tests:range_test                         PASSED in 3.2s
```